src/reports/data/sonata_data.cpp

```diff
@@ -20,48 +19,54 @@ m_total_elements(0), m_remaining_steps(0), m_buffer_size(0), m_steps_to_write(0)
-
+SonataData::SonataData(const std::string& report_name, const std::vector<double>& spike_timestamps, const std::vector<int>& spike_node_ids) {
     m_spike_timestamps = spike_timestamps;
     m_spike_node_ids = spike_node_ids;
     m_io_writer = std::make_unique<HDF5Writer>(report_name);
 }
 
 void SonataData::prepare_buffer(size_t max_buffer_size) {
     logger->trace("Prepare buffer for {}", m_report_name);
+
     for (auto& kv : *m_nodes) {
-        m_total_elements += kv.second.get_num_elements();
+        m_total_elements += kv.second->get_num_elements();
+    }
+
+    // Nothing to prepare as there is no element in those nodes
+    if(m_total_elements == 0) {
+        return;
     }
-    if(m_total_elements > 0) {
-        // Calculate the timesteps that fit given a buffer size
+
+    // Calculate the timesteps that fit given the buffer size
+    {
         int max_steps_to_write = max_buffer_size / sizeof(double) / m_total_elements;
-        if (max_steps_to_write < m_num_steps) {
+        if (max_steps_to_write < m_num_steps) { // More step asked that buffer can contains
             if(max_steps_to_write < ReportingLib::m_min_steps_to_record) {
                 m_steps_to_write = ReportingLib::m_min_steps_to_record;
             } else {
                 // Minimum 1 timestep required to write
-                m_steps_to_write = max_steps_to_write > 0? max_steps_to_write: 1;
+                m_steps_to_write = max_steps_to_write > 0 ? max_steps_to_write: 1;
             }
-        } else {
+        } else { // all the step asked fit into the given buffer
             // If the buffer size is bigger that all the timesteps needed to record we allocate only the amount of timesteps
             m_steps_to_write = m_num_steps;
         }
+    }
 
-        m_remaining_steps = m_num_steps;
+    m_remaining_steps = m_num_steps;
 
-        if(ReportingLib::m_rank == 0) {
-            logger->debug("-Total elements: {}", m_total_elements);
-            logger->debug("-Num steps: {}", m_num_steps);
-            logger->debug("-Steps to write: {}", m_steps_to_write);
-            logger->debug("-Max Steps to write: {}", max_steps_to_write);
-            logger->debug("-Max Buffer size: {}", max_buffer_size);
-        }
-        m_buffer_size = m_total_elements * (m_steps_to_write);
-        m_report_buffer.resize(m_buffer_size);
-        if(ReportingLib::m_rank == 0) {
-            logger->debug("-Buffer size: {}", m_buffer_size);
-        }
+    if(ReportingLib::m_rank == 0) {
+        logger->debug("-Total elements: {}", m_total_elements);
+        logger->debug("-Num steps: {}", m_num_steps);
+        logger->debug("-Steps to write: {}", m_steps_to_write);
+        logger->debug("-Max Buffer size: {}", max_buffer_size);
+    }
+
+    size_t m_buffer_size = m_total_elements * (m_steps_to_write);
+    m_report_buffer.resize(m_buffer_size);
+
+    if(ReportingLib::m_rank == 0) {
+        logger->debug("-Buffer size: {}", m_buffer_size);
```

Comment:
I use m_ for class members.
Since m_buffer_size is not a class member anymore I would call it just buffer_size.